tests/utils/PolygonTest.cpp

```diff
@@ -183,5 +183,57 @@ void PolygonTest::getEmptyHolesTest()
     }
 }
 
+void PolygonTest::simplifyCircle()
+{
+    Polygons circle_polygons;
+    PolygonRef circle = circle_polygons.newPoly();
+    constexpr coord_t radius = 100000;
+    constexpr double segment_length = 1000;
+    constexpr double tau = 6.283185307179586476925286766559; //2 * pi.
+    constexpr double increment = segment_length / radius; //Segments of 1000 units.
+    for (double angle = 0; angle < tau; angle += increment)
+    {
+        circle.add(Point(std::cos(angle) * radius, std::sin(angle) * radius));
+    }
+
+    constexpr coord_t minimum_segment_length = segment_length + 10;
+    circle_polygons.simplify(minimum_segment_length, 999999999); //With segments of 1000, we need to remove exactly half of the vertices to meet the requirement that all segments are >1010.
+    constexpr coord_t maximum_segment_length = segment_length * 2 + 20; //+20 for some error margin due to rounding.
+
+    for (size_t point_index = 1; point_index < circle.size() - 1; point_index++) //Don't check the last vertex. Due to odd-numbered vertices it has to be shorter than the minimum.
+    {
+        coord_t segment_length = vSize(circle[point_index % circle.size()] - circle[point_index - 1]);
+        std::stringstream ss_short;
+        ss_short << "Segment " << (point_index - 1) << " - " << point_index << " is too short! " << segment_length << " < " << minimum_segment_length;
+        CPPUNIT_ASSERT_MESSAGE(ss_short.str(), segment_length >= minimum_segment_length);
+        std::stringstream ss_long;
+        ss_long << "Segment " << (point_index - 1) << " - " << point_index << " is too long! " << segment_length << " > " << maximum_segment_length;
+        CPPUNIT_ASSERT_MESSAGE(ss_long.str(), segment_length <= maximum_segment_length);
+    }
+}
+
+void PolygonTest::simplifyZigzag()
```

Comment:
Nice! More UnitTests! :-) Maybe also add a test where it should _not_ simplify something further?

Reply:
Yes, added two of those! One for each limiting parameter.